### PR TITLE
Refactor diag movements 

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Refactor diag movement code to be true diag movement and not qol like it is right now.
You can now shuffle on diagonals
Allow crusher to diag charge

## Why It's Good For The Game

Shuffling and pushing in diag is really neat. Diag speed = 1.5 is more closer to true diag movement (1.42), and i hope it's not that big of a balance breaker (it should still be possible for a xeno to pursue and attack a marine fleeing, but that should be tested). Can still be put back to two times, without any issue

The refactor prevent silly bugs where you can catch fire when diag moving near lava/fire, prevent double stack fire bug, etc etc.

Allow crusher and bull to charge diagonally, because why not? Because of the slower speed, this will rarely be usefull

## Changelog
:cl:
refactor: Refactor diag movements
balance: Crusher and bull can charge diagonally
qol: Allow to shuffle and push in diagonal movements
/:cl:

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -196,6 +198,7 @@
 			. = ..()
 		else //Diagonal move, split it into cardinal moves
 			moving_diagonally = FIRST_DIAG_STEP
+			olderloc = loc
```
Comment (SplinterGP) [inline code/game/atoms_movable.dm:201]: do you actually need this var at all ? there's oldloc

**Comment by SplinterGP** [comment]
only TM's allowed

File: code/modules/mob/mob_movement.dm [hunk 2]
```diff
@@ -141,7 +141,7 @@
 	. = ..()
 
 	if((direct & (direct - 1)) && mob.loc == n) //moved diagonally successfully
-		add_delay *= 2
+		add_delay *= 1.5
```
Comment (SplinterGP) [inline code/modules/mob/mob_movement.dm:144]: keep this at 2 for now
```suggestion
		add_delay *= 2
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by EgorDinamit** [comment]
Diagonal charge.  :eyes:

**Comment by SplinterGP** [comment]
i'm gonna close this since i'm not going to deal with it and we don't have other maintainers up to reviewing it.

**Pull request closed without merging**